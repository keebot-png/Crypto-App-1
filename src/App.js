// import "./App.css";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { useState } from "react";
import { Header, ProductDetail, ProductListing } from "./components";

function App() {
  // const [searchTerm, setSearchTerm] = useState("");
  // const [products, setProducts] = useState([]);
  // const [searchResults, setSearchResults] = useState([]);

  // const searchHandler = (searchItem) => {
  //   setSearchTerm(searchItem);
  //   if (searchItem !== "") {
  //     const newItemList = products.filter((product) => {
  //       return Object.values(product)
  //         .join(" ")
  //         .toLowerCase()
  //         .includes(searchItem.toLowerCase());
  //     });
  //   }
  // };

  return (
    <div className="App">
      <Router>
        <Header />
        <Routes>
          <Route
            path="/"
            element={
              <ProductListing />
            }
          />
          <Route path="/product/:productId" element={<ProductDetail />} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
