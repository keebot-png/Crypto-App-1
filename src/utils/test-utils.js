import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
// As a basic setup, import your same slice reducers
import reducer from '../redux/products/products';

const render = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {},
) => {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';
export { render };