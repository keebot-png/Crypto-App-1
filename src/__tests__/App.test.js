import React from 'react';
import { render, fireEvent, screen } from '../utils/test-utils';
import App from '../App';

describe('App', () => {
  it('renders page correctly', () => {
    render(<App />);
    const tree = render(<App />);
    expect(tree).toMatchSnapshot();
  });

  it('it loads "PRoducts" on first render', async () => {
    render(<App />);
    expect(await screen.findByText('Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops')).toBeInTheDocument();
  });

  it('navigates to Details and displays product details', async () => {
    render(<App />);
    expect(await screen.findByText('Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops')).toBeInTheDocument();
    fireEvent.click(screen.getByText('Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops'));
    expect(await screen.findByText('Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday')).toBeInTheDocument();
  });
});
