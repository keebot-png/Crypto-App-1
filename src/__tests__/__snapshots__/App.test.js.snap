// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="ui fixed menu"
        >
          <div
            class="ui container center"
          >
            <h2>
              Shopping Store
            </h2>
          </div>
        </div>
        <div
          class="main"
        >
          <input
            placeholder="Filter by product name..."
            type="text"
            value=""
          />
          <div
            class="ui centered grid container"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="App"
      >
        <div
          class="ui fixed menu"
        >
          <div
            class="ui container center"
          >
            <h2>
              Shopping Store
            </h2>
          </div>
        </div>
        <div
          class="main"
        >
          <input
            placeholder="Filter by product name..."
            type="text"
            value=""
          />
          <div
            class="ui centered grid container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="ui fixed menu"
      >
        <div
          class="ui container center"
        >
          <h2>
            Shopping Store
          </h2>
        </div>
      </div>
      <div
        class="main"
      >
        <input
          placeholder="Filter by product name..."
          type="text"
          value=""
        />
        <div
          class="ui centered grid container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
